### PR TITLE
new: support sidecar resource settings in handler

### PR DESCRIPTION
**Pull request by gnosek**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

Replace the specific imageAuthSecret field with a sidecar config that also includes resource settings (essential flag, cpu shares, memory reservation/limit)

### CODE REVIEW

**Comment by gnosek** [comment]
~Note: it's a draft since I haven't updated the cfn template yet. Incoming.~

I'm an idiot. The cfn template lives in a different repo.

**Review by francesco-racciatti — DISMISSED** [review]

File: runtimes/cloudformation/cmd/handler/main.go [hunk 1]
```diff
@@ -78,6 +79,11 @@ func GetConfig() *cfnpatcher.Configuration {
 	disableRepoHints := os.Getenv("KILT_DISABLE_REPO_HINTS")
 	logGroup := os.Getenv("KILT_LOG_GROUP")
 	parameterizeEnvars := os.Getenv("KILT_PARAMETERIZE_ENVARS")
+	sidecarEssential := os.Getenv("KILT_SIDECAR_ESSENTIAL")
+	sidecarCpu := os.Getenv("KILT_SIDECAR_CPU")
+	sidecarMemoryLimit := os.Getenv("KILT_SIDECAR_MEMORY_LIMIT")
+	sidecarMemoryReservation := os.Getenv("KILT_SIDECAR_MEMORY_RESERVATION")
+	sidecarConfig := os.Getenv("KILT_SIDECAR_CONFIG")
```
Comment (pgcrooks-sysdig) [inline runtimes/cloudformation/cmd/handler/main.go:86]: Are these documented anywhere? I know `KILT_LOG_GROUP` etc are documented in `kilt-definitions/cloudformation/serverless-patcher/README.md` already, should they be moved into a README in this repo?

Non-blocking comment.

**Review by pgcrooks-sysdig — DISMISSED** [review]
LGTM!

Shiny new-ness

**Review by francesco-racciatti — APPROVED** [review]

**Review by pgcrooks-sysdig — APPROVED** [review]
LGTM

**Pull request merged**